controllers/cluster_controller.go

```diff
@@ -287,3 +296,80 @@ func splitMachineList(list *clusterv1.MachineList) (*clusterv1.MachineList, *clu
 	}
 	return controlplanes, nodes
 }
+
+func (r *ClusterReconciler) reconcileControlPlaneInitialized(ctx context.Context, cluster *clusterv1.Cluster) error {
+	if cluster.Status.ControlPlaneInitialized {
+		return nil
+	}
+
+	machines, err := r.getMachinesInCluster(ctx, cluster.Namespace, cluster.Name)
+	if err != nil {
+		r.Log.Error(err, "error getting machines in cluster", "cluster", cluster.Name)
+		return err
+	}
+
+	for _, m := range machines {
+		if util.IsControlPlaneMachine(m) && m.Status.NodeRef != nil {
+			cluster.Status.ControlPlaneInitialized = true
+			return nil
+		}
+	}
+
+	return nil
+}
+
+// getMachinesInCluster returns all of the Machine objects that belong to the cluster
+func (r *ClusterReconciler) getMachinesInCluster(ctx context.Context, namespace, name string) ([]*clusterv1.Machine, error) {
+	if name == "" {
+		return nil, nil
+	}
+
+	machineList := &clusterv1.MachineList{}
+	labels := map[string]string{clusterv1.MachineClusterLabelName: name}
+
+	if err := r.Client.List(ctx, machineList, client.InNamespace(namespace), client.MatchingLabels(labels)); err != nil {
+		return nil, errors.Wrap(err, "failed to list machines")
+	}
+
+	machines := []*clusterv1.Machine{}
+	for i := range machineList.Items {
+		m := &machineList.Items[i]
+		if m.DeletionTimestamp.IsZero() {
+			machines = append(machines, m)
+		}
+	}
+	return machines, nil
+}
+
+// machineToCluster is a handler.ToRequestsFunc to be used to enqueue requests for reconciliation
+// for Cluster to update its status.controlPlaneInitialized field
+func (r *ClusterReconciler) machineToCluster(o handler.MapObject) []ctrl.Request {
```

Comment:
Do you think calling this `controlPlaneMachineToCluster` would make it slightly clearer as to the purpose of this function?